In [18]:
import pandas as pd
import numpy as np
import math
from matplotlib import pyplot as plt

## Determine range of plausible nanofiber surface areas 

In [20]:
def nanofiber_surfacearea(radius, concentration):
    '''
    radius = [L]
    concentration = [nanofibers/L] L in axis perpendicular to nanofiber direction
    
    SA of cylinder sides = 2*pi*r*h
    '''
    indiv_sa = 1 * math.pi * radius #[L]

    unit_sa = indiv_sa * concentration #[L/L]
    
    return unit_sa

    

In [63]:
# fiber radii range: .15 - 1 um
steps = 25
fiber_radii = np.linspace(.15, 1, steps) # in microns

# fiber_concentrations = 1/(fiber_radii*2) # in fibers/micron perpendicular

In [66]:
# Fiber concentration cannot be more than 1/(fiber diameter) per unit

concentration_range = []

for i in fiber_radii:
    fiber_concentration = np.linspace(0, 1/(2*i), 10)
    for j in fiber_concentration[1:]:
        concentration_range.append(nanofiber_surfacearea(i, j))

In [70]:
sa_max = max(concentration_range)
sa_min = min(concentration_range)
sa_min, sa_max

(0.17453292519943292, 1.5707963267948968)

## Results:

### Min Surface Area: .175 

### Max Surface Area: 1.571 

#### Units in [SA Nanofibers]/[Unit SA]